### DISCUSSION TITLE
ideas for actual quality control rules

### DISCUSSION THREAD

**Discussion by urapadmin** [arch-kiosk/arch-kiosk-office#1048] [users: ideas for new features]
Here we collect ideas for what we want to have checked by the new quality control in filemaker recording and kiosk.
Please state at least the severity of a violation of a rule: 
-  error: should be rare because the system will be quite clear and annoying about this
- warning: will be clearly indicated but not intrude on the workflow
- hint: perceived rather as a helpful hint but not annoying at all

**Comment by urapadmin** [comment]
* a locus should have an opening date -> error
* a locus should have a type -> error
* a locus should have somebody in recorded by -> error
* a locus should have at least one photo -> warning 
* a locus should have relationships -> warning?
* a locus should have a description -> warning
* if a locus has a closing date it should have at least two locus photos: An opening and a closing shot. -> hint

**Comment by lbestock** [comment]
If a project uses lots and a locus has collected material the locus had damn well better have at least one lot, and every CM had damn well better have one. But it strikes me that this might be on the complicated side.

**Comment by urapadmin** [comment]
> 
> 
> If a project uses lots and a locus has collected material the locus had damn well better have at least one lot, and every CM had damn well better have one. But it strikes me that this might be on the complicated side.

that's a good one. "If a project uses lots" I can ignore,  because if a project does not, it just disables the rule. So we are down to:
- if a locus has collected material it must have a lot -> warning or even error?
- Every collected material must belong to a lot -> error

**Comment by luizaogs** [comment]
Elevations, maybe with hints? Or warnings. If a locus has an opening date it should have an opening elevation, and if a locus has a closing date it should have a closing elevation. (Unless, I suppose, the expectation is that they will be recovered later from models or something. But still.)

**Comment by lbestock** [comment]
For those who use elevations, for sure - good thought. It's one of the most important things and most overlooked. However, I do suspect that our elevation issues are the most obvious hiccup occurring from the fact that we don't yet integrate geospatial data. In the end that isn't a hint to write down a number, it's a hint to call the surveyor over so you get the locus measured properly and into the GIS.

**Comment by lbestock** [comment]
> * if a locus has collected material it must have a lot -> warning or even error?
> * Every collected material must belong to a lot -> error

I think we don't need the first one at all; an error in CM if there is no lot forces the issue.

**Reply by urapadmin** [reply]
yes and no. The one thing would be checked on locus level, the other thing on collected material level. So there is a difference in where the hint would show. Perhaps it is enough to show it in collected material.

**Comment by urapadmin** [comment]
should we get USTP and ARCH1900 on here?

**Reply by urapadmin** [reply]
we decided that we should. Let' see if something like this is more manageable using the discussion feature.

**Reply by lbestock** [reply]
Have there always been reaction emojis? They seem so much more visually prominent to me now. Maybe because I have gone craaaazy. Too crazy to craft a normal person email to USTP and ARCH1900 right now.

**Reply by luizaogs** [reply]
(You think you just went crazy now?)

**Comment by aes1884** [comment]
ARCH1900 considerations:

- Loci need photos (at least one, either opening or closing) - this was a huge problem last year. At least a warning if not an error. Would be great if we could force an opening photo.
- Relations and interpretations were largely ignored. Hints for these.
- We have a big problem with low frequency in creating narratives and dayplans. Hint for these? Maybe when the user logs out?
- much less concerned about anything that has to do with collected materials since the number autopopulates and we usually leave most of the collected material recording for the lab, which is a much more controlled and supervisable recording environment

**Reply by urapadmin** [reply]
to test our approach we have implemented quality control only for loci at this point, so low frequency in dayplans and narratives is postponed (but a sure thing). 
- relations can be checked
- interpretations can be checked
- having at least one locus photo would be the rule we can start with

**Comment by urapadmin** [comment]
Laurel and I came across a complex rule addressing narratives:
there should be a narrative every day by every user. So the rule would check if the logged in user has at least one narrative on the current day. This is also a good example for different rules for kiosk and the recording:
- in the recording system the rule can be as described, the question is only: On what days does it make sense? It seems reasonable to me only on days when the unit has actually been dug. So checking loci activity as well? 
- in kiosk that rule would not work at all because kiosk does not know anything about what users where logged in. Kiosk could only check if there is at least one narrative per day. Here the question arises again: On what day would we expect a narrative?

**Comment by lbestock** [comment]
In the recording system, I think there should be a narrative any time any data at all in a unit is added to or changed. So not just digging. I do this all the time in URAP - if I'm working on an article and figuring out the stratigraphy I'm adding to interpretations in the recording system. And I make a narrative to say I am doing so; that journalling of working in the unit applies to research and intellectual work that affect the recording as well as it does to physical work that affects the recording.

**Reply by urapadmin** [reply]
that is a hard one. "any data at all in a unit" encompasses a lot depending on the scope of that context. So somebody adds a locus -> unit changed? Your "interpretation" is a locus field. Somebody adds a collected material -> unit changed? CMK changes or adds a sherd? Unit changed? Most of your units would be red.

**Reply by lbestock** [reply]
Hmpf the sherd example is instructive, particularly if we say adding a parallel or a note - that doesn't necessarily mean working in the unit, could just be running across something and wanting to make sure it gets into the recording system before one forgets. I don't know, though, any of the others do make me want a narrative. Narratives are often one of the really helpful things when trying to put it all back together again later. If you're adding CM then aren't you working in the unit? Why not say so and say WHY, which is where the narratives are really helpful? I mean I added to the FA narrative when I was in Boston messing around in the Wheeler archives and then I didn't even change any data. It was still relevant.

**Comment by lbestock** [comment]
(Date closed as a QC rule is driving me a little nuts :) It is always on! And a wall is never closed! I am not so sure I will be annoyed with it in the field ⛺️ at the end of a season when it reminds me to check that all locus relations have been made for a locus and whatnot and I only mark it closed when it's really done being recorded for the season, but in testing and in looking at older stuff it makes me unhappy.)

**Reply by urapadmin** [reply]
why is a wall never closed __in that unit__? What does "closing" a locus mean given that we are looking at it as part of what that unit reveals? 
It is that part of the wall in that unit, no? Not the whole wall.

**Comment by lbestock** [comment]
This is such an old friend! SUCH an old friend, the relationship of the wall in the unit to the wall wall. Makes me happy. Makes me want to clarify by saying "locus closed" is nonsense. What is always meant, wall or not, is "recording of this locus is closed for this season". Like a ticket it should be reopenable.

**Reply by lbestock** [reply]
Or even more "field observation and recording of this locus is complete for this season". Not a good label in FM, though.
